tripy/tests/frontend/test_stride.py

```diff
@@ -0,0 +1,68 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import re
+
+import cupy as cp
+import numpy as np
+import torch
+
+import tripy as tp
+from tests.helper import raises
+
+
+class TestStride:
+    def assert_error_message(self, excinfo, tensor_type, expected_suggestion):
+        error_message = str(excinfo.value)
+        assert "Non-canonical strides are not supported for Tripy tensors." in error_message
+        assert f"For {tensor_type}, use {expected_suggestion}" in error_message
+
+    def tripy_byte_order_strides(self, data):
+        return tuple(s * data.dtype.itemsize for s in tp.Tensor(data).stride())
+
+    def test_non_canonical_stride(self):
+        # PyTorch test
+        t_torch = torch.arange(12, dtype=torch.float32).reshape(3, 4)
+        a_torch = t_torch.transpose(0, 1)
+        with pytest.raises(tp.TripyException) as excinfo:
+            tp.Tensor(a_torch)
+        self.assert_error_message(excinfo, "PyTorch Tensor", "tensor.contiguous() or tensor.clone()")
```

Comment:
Could we do:
```suggestion
        with raises(tp.TripyException, "Non-canonical strides are not supported for Tripy tensors"):
            tp.Tensor(a_torch)
```
and then remove the `assert_error_message` helper?

Reply:
Also, maybe instead of checking with torch/numpy/cupy tensors here, we can just unit test `check_tensor_type_and_suggest_contiguous` separately?

Reply:
Agreed.